src/overlays/actors/ovl_En_Go/z_en_go.c

```diff
@@ -1222,486 +1631,614 @@ s32 func_80A13B1C(EnGo* this, PlayState* play) {
+                   (gSaveContext.save.weekEventReg[22] & 0x4 /* Elder's son has been calmed */)) {
+            // While asleep, rhythmicallly snore and distort shape to show breathing.
+            this->actor.scale.x = this->scaleFactor - (Math_SinS(this->snorePhase) * (0.1f * ENGO_NORMAL_SCALE));
+            this->actor.scale.y = this->scaleFactor + (Math_SinS(this->snorePhase) * (0.1f * ENGO_NORMAL_SCALE));
+            this->actor.scale.z = this->scaleFactor + (Math_SinS(this->snorePhase) * (0.1f * ENGO_NORMAL_SCALE));
+
+            if (this->snorePhase == 0) {
+                this->sleepState = -this->sleepState;
+                sfxId = (this->sleepState > ENGO_AWAKE) ? NA_SE_EN_GOLON_SNORE1 : NA_SE_EN_GOLON_SNORE2;
                 Actor_PlaySfxAtPos(&this->actor, sfxId);
             }
-            this->unk_3AE += 0x400;
-            this->actor.shape.yOffset = (this->actor.scale.y / this->unk_3A4) * 14.0f;
-            SubS_UpdateFlags(&this->unk_390, 3, 7);
+            this->snorePhase += 0x400; // ~5.7 degrees per frame or ~1/3 rotation per second.
+            this->actor.shape.yOffset = (this->actor.scale.y / this->scaleFactor) * ENGO_ROLLEDUP_Y_OFFSET;
+            SubS_UpdateFlags(&this->actionFlags, 3, 7);
         }
     } else if ((this->actor.xzDistToPlayer >= 240.0f) || (this->actor.playerHeightRel >= 20.0f) ||
-               (this->unk_3EC != 0)) {
-        func_80A12C48(this, play, 4);
-        this->unk_390 &= ~0x80;
-        this->unk_390 &= ~0x200;
-        this->unk_390 |= 0x4000;
-        this->actor.shape.yOffset = 0.0f;
+               (this->sleepState != ENGO_AWAKE)) {
+        EnGo_ChangeAnim(this, play, ENGO_ANIM_ROLL);
+        this->actionFlags &= ~ENGO_FLAG_STANDING;
+        this->actionFlags &= ~ENGO_FLAG_ROLLEDUP;
+        this->actionFlags |= ENGO_FLAG_ROLLINGUP;
+        this->actor.shape.yOffset = ENGO_STANDING_Y_OFFSET;
     }
 
-    SubS_FillLimbRotTables(play, this->unk_3CE, this->unk_3C8, ARRAY_COUNT(this->unk_3CE));
-    Math_ApproachS(&this->actor.shape.rot.y, sp26, 4, 0x2AA8);
+    SubS_FillLimbRotTables(play, this->limbRotTableY, this->limbRotTableZ, ARRAY_COUNT(this->limbRotTableY));
+    Math_ApproachS(&this->actor.shape.rot.y, targetRot, 4, 0x2AA8);
 }
 
-void func_80A14E14(EnGo* this, PlayState* play) {
-    Actor* actor = this->colliderCylinder.base.ac;
+/**
+ * Action function for a frozen goron
+ *
+ * Transitions to the thawing cutscene
+ */
+void EnGo_Frozen(EnGo* this, PlayState* play) {
+    Actor* actorCollidedWith = this->colliderCylinder.base.ac;
 
-    if ((this->unk_390 & 0x1000) && (((actor != NULL) && (actor->id == ACTOR_OBJ_AQUA) && (actor->params & 1)) ||
-                                     (this->actor.colChkInfo.damageEffect == 2))) {
-        this->actionFunc = func_80A14E74;
+    if ((this->actionFlags & ENGO_FLAG_HIT_BY_OTHER) &&
+        (((actorCollidedWith != NULL) && (actorCollidedWith->id == ACTOR_OBJ_AQUA) &&
+          (actorCollidedWith->params & 1)) ||
```

Comment:
This has a define, although it should be better-named, it's when it's hot spring water?

Reply:
`#define AQUA_HOT(thisx) ((thisx)->params & 1)`
This, in `z_obj_aqua.h`